### ISSUE TITLE
Android client block hotspot connection

### ISSUE THREAD

**Issue by linsui** [tailscale/tailscale#4160]
### What is the issue?

With tailscale enabled on Android, devices connected to the hotspote can't access network.

### Steps to reproduce

Enable tailscale on Android then connect to its hotspot.

### OS

Android

### Bug report

BUG-f32c7dcf9e667c808c7401507f0e99d53a1416158566c46c59d9c2457a238c8d-20220313144438Z-f16eb40d0d1e407d

**Comment by bradfitz** [comment]
Which Android version, which phone?

**Comment by linsui** [comment]
Android 11 LineageOS 18.1 Xiaomi Mi 6.

**Comment by bradfitz** [comment]
I'm not familiar with how LineageOS handles the routing table for wifi hot spots.

At least in stock/Google Android, the wifi hot spot is an entirely separating table from what your apps on your phone see. (So apps on your phone would go over Tailscale, but people using your wifi hot spot would not.)

It's possible that functionality is broken on LineageOS.

**Comment by linsui** [comment]
LineageOS has a option to allow hotspot client using VPN. Tailscale and hotspot works well if the opthon is disabled. Other VPN apps work well with hotspot. And the traffic from hotspot goes through those VPNs.

**Comment by ignoramous** [comment]
LineageOS likely mucks with ip / nf to enable such functionality. (root) Apps such as MyGod's VPNHotSpot most certainly do: https://github.com/Mygod/VPNHotspot/blob/master/mobile/src/main/java/be/mygod/vpnhotspot/net/Routing.kt

**Comment by DentonGentry** [comment]
> LineageOS has a option to allow hotspot client using VPN. Hotspot works well with tailscale if the opthon is disabled. Other VPN apps work well with the option and the traffic from hotspot goes through those VPNs.

Does this mean that this feature request is already met if LineageOS was compiled to allow hotspot to use the VPN?

**Comment by linsui** [comment]
> Does this mean that this feature request is already met if LineageOS was compiled to allow hotspot to use the VPN?

No. It's an option which can be enabled in the setting to allow other device using the VPN through the hotspot. The option works with other VPN but not Tailscale. But if Tailscale is enabled, the device on hotsport can't connect to the internet at all.
